docusaurus/docs/packages/pkg/relayer/relayminer_config.md

```diff
@@ -0,0 +1,379 @@
+  - [`query_node_grpc_url`](#query_node_grpc_url)
+  - [`tx_node_grpc_url`](#tx_node_grpc_url)
+- [RelayMiner proxies](#relayminer-proxies)
+  - [`name`](#name)
+  - [`type`](#type)
+  - [`host`](#host)
+- [Suppliers](#suppliers)
+  - [`name`](#name-1)
+  - [`type`](#type-1)
+  - [`service_config`](#service_config)
+    - [`url`](#url)
+    - [`authentication`](#authentication)
+    - [`headers`](#headers)
+  - [`hosts`](#hosts)
+  - [`proxy_names`](#proxy_names)
+- [Examples](#examples)
+  - [Proxy -\> Supplier referencing](#proxy---supplier-referencing)
+  - [Full config example](#full-config-example)
+
+# Usage
+
+The `RelayMiner` start command accepts a `--config` flag that points to a configuration
+`yaml` file that will be used to setup the `RelayMiner` instance.
+
+```bash
+poktrolld relayminer --config ./relayminer_config.yaml --keyring-backend test
+```
+
+# Structure
+
+The `RelayMiner` configuration file is a `yaml` file that contains `top level options`,
+`proxies` and `suppliers` sections.
+
+# Top level options
+
+```yaml
+signing_key_name: <string>
+smt_store_path: <string>
+```
+## `signing_key_name`
+ _`Required`_
+
+The name of the key that will be used to sign transactions, derive the public key
+and the corresponding address. This key name must be present in the keyring that is used
+to start the `RelayMiner` instance.
+
+## `smt_store_path`
+_`Required`_
+
+The relative or absolute path to the directory where the `RelayMiner` will store
+the `SparseMerkleTree` data on disk. This directory is used to persist the `SMT`
+in a BadgerDB KV store data files.
+
+# Pocket node connectivity
+
+```yaml
+pocket_node:
+  query_node_rpc_url: tcp://<host>
+  query_node_grpc_url: tcp://<host>
+  tx_node_grpc_url: tcp://<host>
+```
+## `query_node_rpc_url`
+_`Required`_
+
+The RPC URL of the Pocket node that allows the `RelayMiner` to subscribe to events
+via websockets. It is then re-formatted as `ws://<host>/websocket` and establishes a
+persistent connection to the Pocket node to stream events such as latest blocks,
+application staking events, etc...
+
+## `query_node_grpc_url`
+_`Optional`_
+
+The gRPC URL of the Pocket node that allows the `RelayMiner` to query/pull data from
+the Pocket network (eg. Sessions, Accounts, etc...). If unspecified, `tx_node_grpc_url`
+value will be used.
+
+## `tx_node_grpc_url`
+_`Required`_
+
+The gRPC URL of the Pocket node that allows the `RelayMiner` to submit transactions.
+It may have a different host than the `query_node_rpc_url` or the `tx_node_grpc_url`
+but same values are acceptable too.
+
+# RelayMiner proxies
+
+The `proxies` section of the configuration file is a list of proxies that the
+`RelayMiner` will use to start servers by listening on the configured host.
+
+At least one proxy is required for the `RelayMiner` to start.
+
+```yaml
+proxies:
+  - name: <string>
+    type: <enum{http}>
+    host: <host>
+```
+
+## `name`
+_`Required`_, _`Unique`_
+
```

Comment:
`TODO_CONSIDERATION: Rename to supplier_name`

Wdyt?

Reply:
I think it should be `proxy_name`, In that case, yes, I'm for renaming it.

Reply:
I propose we rename this to `proxy_name` and the `suppliers[i].name` to `service_id`

Reply:
+1 to both